Add EVALS_CODE venv using ElutherAI's original lm-evaluation-harness repository

arobergeTT opened the pull request:
### Summary

Adds a dedicated virtual environment (`EVALS_CODE`) for code evaluation benchmarks (`MBPP`, `HumanEval`) using ElutherAI's original lm-evaluation-harness repository.

### Changes
* `workflows/workflow_types.py` : Added `EVALS_CODE` enum to `WorkflowVenvType`
* `workflows/workflow_venvs.py`: Added `setup_evals_code()` function that:
* Installs lm-evaluation-harness with code evaluation extensions
* Pins critical packages (protobuf, pyjwt==2.7.0, pillow==11.0.0, datasets==3.1.0)
* `evals/run_evals.py` : Auto-applies security flags (`--trust_remote_code`, `--confirm_run_unsafe_code`) for code evaluation tasks

### Key Features
* Isolation : Dedicated environment prevents dependency conflicts
* Security : Automatic security flags for code execution
* Optimized : Tailored dependencies for code evaluation tasks

Ready for enabling code evaluation tasks in `eval_config.py`.

arobergeTT committed:
Create new venv for coding tasks - i.e. MBPP, HumanEval

arobergeTT committed:
Set os.environ[HF_ALLOW_CODE_EVAL] = 1 in the current process before any lm_eval modules are imported

stisiTT commented on evals/run_evals.py:
```diff
@@ -189,6 +194,18 @@ def main():
+    # Check if any tasks require code evaluation and set environment variable early
+    # This must be set in os.environ (not just env_vars) because lm_eval modules
+    # check for it during their import phase, before the subprocess is fully set up
+    has_code_eval_tasks = any(task.confirm_run_unsafe_code for task in eval_config.tasks)
```
@arobergeTT this did fail when attempting to run and is also the fix that Claude suggests.

arobergeTT replied:
@stisiTT I made appropriate changes. Could you try testing it again?

arobergeTT committed:
Set env_vars[HF_ALLOW_CODE_EVAL] = 1 for coding evals

arobergeTT committed:
Merge branch 'dev' into adam/eval_coder

stisiTT commented:
[run_2025-08-25_17-49-59_id_tt-transformers_Qwen2.5-Coder-32B-Instruct_t3k_evals.log](https://github.com/user-attachments/files/21976291/run_2025-08-25_17-49-59_id_tt-transformers_Qwen2.5-Coder-32B-Instruct_t3k_evals.log)
Pretty consistently running into problems here when evals are being run. Can you take a look at this log file and [my branch](https://github.com/tenstorrent/tt-inference-server/pull/549) to see if I'm setting anything up wrong here?

stisiTT committed:
Adding Qwen2.5-32B-Coder to CODE_EVALS (#549)

Add Qwen2.5-Coder-32B-Instruct model with evaluation tasks

- Added model configuration and performance metrics
- Configured evaluation tasks: HumanEval, MBPP, and livecodebench  
- Updated scoring metrics and batch sizes for all tasks
- Refactored report generation logic

stisiTT committed:
Refactor evaluation tasks in eval_config.py

- Renamed tasks from "mbpp" to "mbpp_instruct" and "humaneval" to "humaneval_instruct"
- Updated scoring metrics to include new result keys
- Removed unused model_kwargs and added gen_kwargs for generation parameters
- Cleaned up the configuration for better clarity and performance

stisiTT committed:
Refactor report generation in run_reports.py

- Updated the structure of the results processing to handle nested task groups
- Improved clarity by renaming variables for better understanding of the data flow

stisiTT committed:
Update evaluation configurations in eval_config.py

- Set published scores and references for MBPP and Humaneval tasks
- Added evaluation task "livecodebench" with scoring metrics and configuration
- Ensured consistency in generation parameters across tasks

stisiTT committed:
Commenting out LiveCodeBench due to indefinite hangs

stisiTT committed:
Merge branch 'dev' into adam/eval_coder

stisiTT committed:
Merge branch 'dev' into adam/eval_coder

# Conflicts:
#	evals/run_evals.py

stisiTT committed:
Adding code execution trust

stisiTT committed:
Merge branch 'dev' into adam/eval_coder

# Conflicts:
#	evals/eval_config.py
